File: BEST-App-main/lib/model/order.dart
class Order {
  late int id;
  late int userId;
  late int paymentId;
  late String address;
  late double total;
  late String createdAt;
  late String phone;
  late int status;

  Order(this.id, this.userId, this.paymentId, this.address, this.total,
      this.createdAt, this.phone,this.status);

  static Map<String, dynamic> toJson(Order order) {
    return {
      'id': order.id,
      'user_id': order.userId,
      'payment_id': order.paymentId,
      'address': order.address,
      'total': order.total,
      'created_at': order.createdAt,
      'phone': order.phone,
      'status': order.status,
    };
  }
}

File: spring-best/src/main/java/com/best/springbest/model/Order.java
package com.best.springbest.model;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.format.annotation.DateTimeFormat;

import javax.persistence.*;
import java.util.Date;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "user_id",nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "payment_id",nullable = false)
    private Payment payment;

    private String address;

    private double total;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    private Date created_at;

    private String phone;

    private int status;

}
